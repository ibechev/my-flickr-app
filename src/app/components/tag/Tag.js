import React, { useRef } from "react";
import PropTypes from "prop-types";

const Tag = ({ index, removeTag, value }) => {
  const indexRef = useRef(index);
  indexRef.current = index;

  const handleRemoveTag = useRef(() => {
    removeTag(indexRef.current);
  });

  return (
    <li className="tag">
      <p className="tag-name">{value}</p>
      <button type="button" onClick={handleRemoveTag.current}>
        <i className="fas fa-trash-alt" />
      </button>
    </li>
  );
};

Tag.propTypes = {
  index: PropTypes.number.isRequired,
  removeTag: PropTypes.func.isRequired,
  value: PropTypes.string.isRequired
};

export default Tag;
