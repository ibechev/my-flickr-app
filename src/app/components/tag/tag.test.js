import React from "react";
import renderer from "react-test-renderer";
import Tag from "./Tag";
import { propTypeErrors } from "../../../../tests/testUtils";
import { mount } from "enzyme";

describe("Expect <Tag />", () => {
  const defaultProps = {
    index: 2,
    removeTag: () => {},
    value: "Tag name"
  };

  it("to render correctly", () => {
    const wrapper = renderer.create(<Tag {...defaultProps} />).toJSON();
    expect(wrapper).toMatchSnapshot();
  });

  it("to not throw error with correct props", () => {
    propTypeErrors(<Tag {...defaultProps} />, defaultProps);
  });

  it("to call prop 'removeTag', when <button /> is clicked", () => {
    const testProps = {
      ...defaultProps,
      removeTag: jest.fn()
    };
    const wrapper = mount(<Tag {...testProps} />);
    wrapper.find("button").simulate("click");
    expect(wrapper.props().removeTag).toHaveBeenCalledTimes(1);
    expect(wrapper.props().removeTag).toHaveBeenCalledWith(testProps.index);
  });
});
