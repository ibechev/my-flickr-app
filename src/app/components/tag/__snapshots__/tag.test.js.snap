// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expect <Tag /> to render correctly 1`] = `
<li
  className="tag"
>
  <p
    className="tag-name"
  >
    Tag name
  </p>
  <button
    onClick={[Function]}
    type="button"
  >
    <i
      className="fas fa-trash-alt"
    />
  </button>
</li>
`;
